src/system/SystemMutex.h

```diff
@@ -163,5 +167,22 @@ inline void Mutex::Unlock(void)
 }
 #endif // CHIP_SYSTEM_CONFIG_FREERTOS_LOCKING
 
+#if CHIP_SYSTEM_CONFIG_ZEPHYR_LOCKING
+inline CHIP_ERROR Mutex::Init(Mutex & aMutex)
+{
+    return System::MapErrorZephyr(k_mutex_init(&aMutex.mZephyrMutex));
+}
+
+inline void Mutex::Lock()
+{
+    k_mutex_lock(&mZephyrMutex, K_FOREVER);
+}
+
+inline void Mutex::Unlock(void)
+{
+    k_mutex_unlock(&mZephyrMutex);
```

Comment:
Should this check that 0 was returnedfrom k_mutex_unlock and die if not?  Or does that not matter much in practice?

Reply:
I can see that other implementations don't check it, either, but I guess I can add a check.